docs/reference/snapshot-restore/restore-snapshot.asciidoc

```diff
@@ -673,3 +693,82 @@ snapshot has the same number of primary shards. This error indicates the index
 in the snapshot has a different number of primary shards.
 
 To resolve this error, try one of the methods in <<restore-index-data-stream>>.
+
+[discrete]
+==== [<repo>:<snapshot-id>] the snapshot was created with {es} version [<x.y.z>] which is higher than the version of this node [<x1.y1.z1>].
+
+You can't restore a snapshot to an earlier version of {es}. The snapshot may
+contain mappings, index settings, and other features that aren't supported in
+the earlier version.
+
+To resolve this error, restore the snapshot to a compatible cluster version. See
+<<snapshot-cluster-compatibility>>.
+
+[discrete]
+==== The index [<index>/<index-id>] was created with version [<x.y.z>] but the minimum compatible version is [<x1.y1.z1>]. It should be re-indexed in {es} <x1.y1> before upgrading to <x2.y2.z2>.
+
+include::index.asciidoc[tag=index-compat]
+
+include::index.asciidoc[tag=restore-reindex-workaround]
+
+[discrete]
+==== index, alias, and data stream names need to be unique, but the following duplicates were found [<alias>(alias of [<index>/<index-id>]]
+
+You can't restore an alias with the same name as an existing index or data
+stream. To resolve this error, you can <<restore-exclude-aliases,exclude
+aliases>> from the restore request. Alternatively, if you no longer need it, you
+can <<delete-restore,delete the existing index or data stream>> and retry the
+restore request.
```

Comment:
Technically, a naming conflict between a data stream and an alias won't generate an error due to https://github.com/elastic/elasticsearch/issues/80972. This assumes that bug will be fixed.